crates/fda/src/main.rs

```diff
@@ -584,6 +588,50 @@ async fn pipeline(action: PipelineAction, client: Client) {
                     .expect("Failed to serialize pipeline stats")
             );
         }
+        PipelineAction::Logs { name, watch } => {
```

Comment:
shouldn't this handle the situation when the pipeline isn't running, and hence there are no logs.  And in the watch mode it should exit without an error when the pipeline terminates?

Reply:
> shouldn't this handle the situation when the pipeline isn't running

```
❯ cargo run -- log --watch test
     Running `/home/gz/workspace/feldera/target/debug/fda log --watch asof`
LOG STREAM UNAVAILABLE: the pipeline has likely not yet started
❯
```

>  And in the watch mode it should exit without an error when the pipeline terminates

```
2024-09-29 01:14:51 INFO [pipeline-019229e6-8e28-7ef1-880e-cfab04c94d6c] Started HTTP server on port 40119
2024-09-29 01:14:51 INFO [pipeline-019229e6-8e28-7ef1-880e-cfab04c94d6c] Pipeline initialization complete
LOG STREAM END: pipeline is being shutdown

❯ echo $?
0
```

did you try it in some way where it didnt?